proposal/function-units.md

```diff
@@ -0,0 +1,175 @@
+# Function Units: Draft 1
+
+*([Issue](https://github.com/sass/sass/issues/3374))*
+
+This proposal restricts the use of invalid units in built-in Sass functions.
+
+## Table of Contents
+
+* [Background](#background)
+* [Summary](#summary)
+  * [Decision Decisions](#decision-decisions)
+    * [Alpha Units](#alpha-units)
+* [Functions](#functions)
+  * [`color.adjust()`](#coloradjust)
+  * [`color.scale()`](#colorscale)
+  * [`color.mix()`](#colormix)
+  * [`list.nth()` and `list.set-nth()`](#listnth-and-listset-nth)
+* [Deprecation Process](#deprecation-process)
+  * [`color.adjust()`](#coloradjust-1)
+  * [`color.scale()`](#colorscale-1)
+  * [`color.mix()`](#colormix-1)
+  * [`list.nth()` and `list.set-nth()`](#listnth-and-listset-nth-1)
+
+## Background
+
+> This section is non-normative.
+
+Sass functions added early during Sass's lifetime were generally quite
+permissive, allowing numbers with units to be passed even if the units were
+ignored. However, this is often quite confusing in practice; for example, a user
+might expect that `color.adjust($color, $alpha: -1%)` would return a color with
+`0.01` less alpha. In fact it returns a transparent color, because the unit is
+ignored entirely.
+
+In some cases, CSS has even added support for units to functions after the
+fact—including support for [percentages in alpha values]. In that case, Sass has
+been forced to deprecate passing incorrect units to those functions before
+adding support for new units.
+
+[percentages in alpha values]: https://www.w3.org/TR/css-color-4/#typedef-alpha-value
+
+## Summary
+
+> This section is non-normative.
+
+This proposal adds the following restrictions:
+
+* The `$alpha` parameter to `color.adjust()`, `color.change()`, `opacify()`,
+  `fade-in()`, `transparentize()`, and `fade-out()` may be either unitless or
+  use the `%` unit. If it uses `%`, it's divided by `100%` before using it.
+
+* The `$weight` parameter to `color.mix()` must have unit `%`.
+
+* The `$n` parameter to `list.nth()` and `list.set-nth()` may not have units.
+
+### Decision Decisions
```

Comment:
Is there a reason for the repetition here or is this supposed to say something else?

Reply:
It's supposed to be "Design Decisions" :laughing: